src/limb.rs

```diff
@@ -84,7 +87,6 @@ impl <'a>Range<'a> {
                 try!(rng.fill(&mut dest_as_bytes));
             }
 
-            let bits_to_mask = self.leading_zero_bits();
             if bits_to_mask > 0 {
                 let keep: Limb = (1 << (LIMB_BITS - bits_to_mask)) - 1;
```

Comment:
I would prefer to name `keep` as `mask` as `mask` seems clearer to me. Also, why did you pull bits_to_mask out of the loop, but not  the calculation of `keep`? I think both should be pulled out of the loop. It won't affect performance but it would improve clarity.